MapboxNavigation/NavigationViewController.swift

```diff
@@ -327,30 +298,24 @@ public class NavigationViewController: UIViewController, RouteMapViewControllerD
         
         super.init(nibName: nil, bundle: nil)
         
-        self.styles = styles ?? [DayStyle(), NightStyle()]
+        self.routeController = RouteController(along: route, directions: directions, locationManager: locationManager ?? NavigationLocationManager())
+        self.routeController.usesDefaultUserInterface = true
+        self.routeController.delegate = self
+        
+        self.directions = directions
+        self.route = route
         
         addChildViewController(mapViewController)
         mapViewController.view.translatesAutoresizingMaskIntoConstraints = false
         view.addSubview(mapViewController.view)
         
-        let v = mapViewController.view!
-        v.topAnchor.constraint(equalTo: view.topAnchor).isActive = true
-        v.leftAnchor.constraint(equalTo: view.leftAnchor).isActive = true
-        v.bottomAnchor.constraint(equalTo: view.bottomAnchor).isActive = true
-        v.rightAnchor.constraint(equalTo: view.rightAnchor).isActive = true
-        
-        self.directions = directions
-        self.route = route
+        self.styleManager = StyleManager(self)
+        self.styleManager.styles = styles ?? [DayStyle(), NightStyle()]
         
-        self.routeController = RouteController(along: route, directions: directions, locationManager: locationManager ?? NavigationLocationManager())
-        self.routeController.usesDefaultUserInterface = true
-        self.routeController.delegate = self
-
+        mapViewController.view!.pinInSuperview()
```

Comment:
This sounds scary, why is this necessary?

Reply:
This is just to set up top|left|bottom|right constraints. Same as before but I moved it to an extension on UIView.

https://github.com/mapbox/mapbox-navigation-ios/blob/1a7392048c6ca63582d88297450312e8304db581/MapboxNavigation/UIView.swift#L39-L45